### PR TITLE
CI fix changes listing and tag name

### PR DESCRIPTION
**Pull request by mathiasvr**
This fixes the missing changes listing in the release notes of the nightly build and adds date to the release tag.

I discussed with @McGiverGim how I was hoping to change the way releases are tagged to e.g. `build-123` or `v10.7.2-123` instead of being based on dates, which is already part of the tag and release. However, unfortunately it turned out to not be simple to preserve release ordering in the web listing and probably requires removing or renaming all the existing tags.
So instead this PR also brings back the date based tags.

### CODE REVIEW

**Comment by mathiasvr** [comment]
@blckmn Can you update the branch protection rules to use the new CI test and build jobs?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]
Approved until we find a way to change the tag format...

**Comment by mathiasvr** [comment]
Seems like [release ordering is buggy](https://github.community/t/release-page-not-sorting-correctly/121002) in general.

**Review by blckmn — APPROVED** [review]

**Pull request merged**
